### PR TITLE
Docs: Add release process doc

### PR DESCRIPTION
**Pull request by matej-g**
Signed-off-by: Matej Gera <matejgera@gmail.com>

We've been talking and preparing the initial release with @saswatamcode and we wanted to codify the process for the future.

The aims I'm following with the release processes are:
1. It should reflect the state of project, i.e. we're in the initial stage where we iterate often in tight loops and have small number of users and contributors. The release should therefore be easy and straightforward.
2. We should be able to rely on `main` branch and tag our releases simply to `main` branch, instead of creating new branches.
3. We should not need release candidate with the current state of project. RC for the initial release is an exception.

### CODE REVIEW

File: docs/release.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Release process
+
+We use semantic versioning for `obstctl` in format `vX.Y.Z`.
```
Comment (saswatamcode) [inline docs/release.md:3]:
```suggestion
We use semantic versioning for `obsctl` in format `vX.Y.Z`.
```

**Review by saswatamcode — APPROVED** [review]
Thanks! LGTM!

Minor nit.

File: docs/release.md (hunk 1 above)
Comment (matej-g) [reply docs/release.md:3]: Funnily enough, 'obst' means 'fruit' in German :banana: :apple: :grapes:, but we cannot control fruits with CLI (yet).

**Review by saswatamcode — APPROVED** [review]

**Review by douglascamata — APPROVED** [review]

File: docs/release.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+1. Create a PR against `main` to update the `version/version.go` to the appropriate next version (and remove the `-dev` suffix).
+   - Normally, with the current (early) state of the project, we always target the next minor version (e.g. we bump from `v0.1.0` to `v0.2.0`), unless there is a compelling reason to release a **patch** version.
+   - We also usually do not have release candidates, with the exception of the initial release.
+2. Once the PR from 1. is approved and merged, start drafting a new release from the [GitHub web UI](https://github.com/observatorium/obsctl/releases/new).
+3. Prepare a new **tag** that will be created together with release. Do this by clicking on `Choose a tag` -> put the new tag into the input field in format `vX.Y.Z`.
+4. Prepare the release title and description. The release title should correspond to the release version. The description can be generated by `Auto-generate release notes`. If the release includes any significant changes, feel free to add a one-sentence summary at the top of the description. If there are any noisy or redundant entries in the auto-generated list, feel free to remove them.
```
Comment (squat) [inline docs/release.md:13]:
```suggestion
4. Prepare the release title and description. The release title should correspond to the release version. The description can be generated by clicking on `Auto-generate release notes`. If the release includes any significant changes, feel free to add a one-sentence summary at the top of the description. If there are any noisy or redundant entries in the auto-generated list, feel free to remove them.
```

**Review by squat — APPROVED** [review]
small nit, otherwise LGTM

**Review by bwplotka — COMMENTED** [review]
👍🏽

**Pull request merged**